docs/grpc_tutorial.md

```diff
@@ -0,0 +1,265 @@
+# gRPC tutorial
+This tutorial explains how to set up the gRPC learner server.
+It assumes that you can already run colearn locally, and that you have already defined your own models and dataloaders 
+(if you're going to do so).
+If you haven't done this then see the tutorials in the Getting Started section.
+
+## Architecture of colearn
+There are two main parts to a collective learning system: the learner and the backend.
+The backend controls the learner, and manages networking, communication with other collective learning nodes, IPFS and smart contracts. 
+The learner is the part that executes machine learning code. 
+This consists of proposing, evaluating and accepting new weights as detailed in the Machine Learning Interface.
+The learner and the backend communicate via gRPC; 
+the learner runs a gRPC server, and the backend runs a gRPC client that makes requests of the learner.
+This separation means that the learner can run on specialised hardware, e.g. a compute server, and does not need to be co-located with the backend.
+
+## Architecture of gRPC server
+The gRPC interface is defined in colearn_grpc/proto/interface.proto.
+This defines the functions that the gRPC server exposes and the format for messages between the server and the client.
+
+gRPC server has a mli factory, and it uses the mli factory to make the learner
```

Comment:
This sentance could possibly be "gRPC server has a mli factory, and it uses this factory to produce an object adhering to the mli interface" ?